python/tank/platform/engine.py

```diff
@@ -1420,6 +1425,61 @@ def _emit_log_message(self, handler, record):
         """
         # default implementation doesn't do anything.
 
+    def _ensure_core_fonts_loaded(self):
+        """
+        Loads the Shotgun approved fonts that are bundled with tk-core.
+
+        This method ensures that the Shotgun approved fonts bundled with core
+        are loaded into Qt's font database. This allows them to be used by apps
+        for a consistent look and feel.
+
+        If a QApplication exists during engine initialization, it is not
+        necessary to call this method. Similarly, subclasses that make use of
+        core's bundled dark look and feel will have the bundled fonts loaded
+        automatically.
+        """
+
+        from sgtk.platform.qt import QtGui
+
+        # if the fonts have been loaded, no need to do anything else
+        if self.__fonts_loaded:
+            return
+
+        if not QtGui.QApplication.instance():
+            # there is a QApplication, so we can load fonts.
+            return
+
+        # fonts dir in the core resources dir
+        fonts_parent_dir = self.__get_platform_resource_file("fonts")
+
+        # in the parent directly, get all the font-specific directories
+        for font_dir_name in os.listdir(fonts_parent_dir):
+
+            # the specific font directory
+            font_dir = os.path.join(fonts_parent_dir, font_dir_name)
+
+            if os.path.isdir(font_dir):
+
+                # iterate over the font files and attempt to load them
+                for font_file_name in os.listdir(font_dir):
+
+                    # only process actual font files. It appears as though .ttf
+                    # is the most common extension for use on win/mac/linux so
+                    # for now limit to those files.
+                    if not font_file_name.endswith(".ttf"):
+                        continue
+
+                    # the actual font file
+                    font_file = os.path.join(font_dir, font_file_name)
+
+                    # load the font into the font db
+                    if QtGui.QFontDatabase.addApplicationFont(font_file) == -1:
```

Comment:
Loading directly from disk. There seems to be a problem on OSX when loading fonts from resources: https://bugreports.qt.io/browse/QTBUG-30917

This is why tk-desktop isn't properly displaying **Open Sans** on OSX!